### ISSUE TITLE
GPU support ?

### ISSUE THREAD

**Issue by eddelbuettel** [grantmcdermott/codespaces-r2u#6]
Can we base a devcontainer on something ubuntu-latest, opening the door to r2u, and also offer GPU / CUDA support?

@eitsupi Is there an existing 'feature' to turn on or how would you suggest we go about this?

My tweet / toot / skeet storm about the r2u codespace/devcontainer worked quite well.  Thank you both for all the help in making it possible with r2u.

**Comment by eitsupi** [comment]
See <https://github.com/devcontainers/features/tree/main/src/nvidia-cuda>

But I haven't use that.

**Comment by eddelbuettel** [comment]
Yeah I am aware of some 'in theory ...' options, but I am eagerly awaiting some 'in practice' validation. 

@grantmcdermott Any pointers?

**Comment by eddelbuettel** [comment]
Also saw another R user have a (different) .devcontainer setup (some "R Data Science Environment" approach for R and also Jupyter)  in a repo, tried that and _of course_ it promptly failed to build.  :disappointed:

**Comment by eitsupi** [comment]
> Also saw another R user have a (different) .devcontainer setup (some "R Data Science Environment" approach for R and also Jupyter) in a repo, tried that and _of course_ it promptly failed to build. 😞

Does it have anything to do with this issue?

**Comment by eddelbuettel** [comment]
No only in the context that 'other approaches exist, I have a sample of size one showing they fail'.

We do better work, generally.  Now, can we best ourselves and add GPU support?  That is this issue ticket.

**Comment by grantmcdermott** [comment]
Hmmm. I'm in a similar boat to you two. I know about the cuda containers---and could have sworn that I saw a GPU codespaces demo a year or so back---but can't think of any real-life repos rn.

(Relatedly: my academic affiliation has just lapsed so I'm a little gun shy for trying expensive GPU-based instances until my courtesy appointment is confirmed by my uni.)
